sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -216,8 +232,13 @@ class CSVOptions(
     format.setDelimiter(delimiter)
     format.setQuote(quote)
     format.setQuoteEscape(escape)
+    lineSeparator.foreach {sep =>
+      format.setLineSeparator(sep)
+      format.setNormalizedNewline(0x00.toChar)
```

Comment:
I know we have some problems here for setting newlines more then 1 character because `setNormalizedNewline` only supports one character. 

This is related with https://github.com/apache/spark/pull/18581#issuecomment-314037750 and https://github.com/uniVocity/univocity-parsers/issues/170

That's why I thought we can only support this for single character for now.